CityOfZion/neo-python
# Accept incoming p2p connections

**What current issue(s) does this address, or what feature is it adding?**
FIx https://github.com/CityOfZion/neo-python/issues/522

**How did you solve this problem?**
Add endpoint for incoming connections, implemented first part of `RemoteNode.StartProtocol()` routine up to `verack` [checking](https://github.com/neo-project/neo/blob/master/neo/Network/RemoteNode.cs#L442)

**How did you make sure your solution works?**
manual testing

**Are there any special changes in the code that we should be aware of?**
Yes. I'm aware that our NeoNode is implemented differently that the C#'s equivalent `RemoveNode` due to historical performance issues. Due to the way it's structured now I had to squeeze in the `Version` and `Verack` payload sending in different locations unlike the `StartProtocol` routine in C#. I also did not go as far as implementing the sequence for `getheaders` ([here](https://github.com/neo-project/neo/blob/master/neo/Network/RemoteNode.cs#L447))  and `getblocks` as we already can update our own state using the peers that we connect to. It would keep the code a bit simpler. Interested to hear if this should still be added

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [X] Did you run `make lint`?
- [X] Did you run `make test`?
- [X] Are you making a PR to a feature branch or development rather than master?
- [X] Did you add an entry to `CHANGELOG.rst`? (if not, please do)
